client/network/src/service.rs

```diff
@@ -814,7 +817,7 @@ where
 	/// a stale fork missing.
 	/// Passing empty `peers` set effectively removes the sync request.
 	fn set_sync_fork_request(&self, peers: Vec<PeerId>, hash: B::Hash, number: NumberFor<B>) {
-		let _ = self.to_worker.unbounded_send(ServiceToWorkerMsg::SyncFork(peers, hash, number));
+		self.chain_sync_service.set_sync_fork_request(peers, hash, number);
 	}
```

Comment:
For now yes but it will be called in other places as well soon, for example in `announce_block()`. The reason I would like to do it this way (having `NetworkService` temporarily call `ChainSync`) is that it allows to refactor the code first out of `Protocol` and then out of `sc-network` entirely. This would allow me to first fix the code inside `sc-network` (extracting the protocol to `sc-network-sync`) and when that's done, only then address the issues in BEEFY/GRANDPA as the network traits they're using need to be decoupled into networking and syncing-related services. 

The current plan I have is that `ChainSync` will continue to act as the provider for these syncing-related services which are still exposed through the `sc-network` crate by `NetworkWorker` but now the calls don't have to be routed through `Protocol`. When all of syncing-code is in `sc-network-sync` (except for these call delegations), then I would move these traits there as well and remove the sync dependency from `sc-network` entirely.

Reply:
Fine by me.